source/logrepl/snapshot_test.go

```diff
@@ -24,28 +24,100 @@ import (
 	"github.com/conduitio/conduit-connector-postgres/test"
 	sdk "github.com/conduitio/conduit-connector-sdk"
 	"github.com/jackc/pgx/v4/pgxpool"
-
 	"github.com/matryer/is"
 )
 
-func TestLifecycle(t *testing.T) {
+var (
+	columns = []string{"id", "key", "column1", "column2", "column3"}
+	key     = "key"
+)
+
+func TestSnapshotIterator_Teardown(t *testing.T) {
 	is := is.New(t)
-	ctx := context.Background()
+	type args struct {
+		ctx context.Context
+	}
+	tests := []struct {
+		name    string
+		setup   func(t *testing.T) *SnapshotIterator
+		args    args
+		wantErr bool
+		wanted  error
+	}{
+		{
+			name: "should return interrupt when next never called",
+			setup: func(t *testing.T) *SnapshotIterator {
+				s, _ := createTestSnapshotIterator(t, columns, key)
+				return s
+			},
+			args: args{
+				ctx: context.Background(),
+			},
+			wantErr: true,
+			wanted:  ErrSnapshotInterrupt,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := tt.setup(t)
+			if err := s.Teardown(tt.args.ctx); (err != nil) != tt.wantErr {
+				if tt.wantErr {
+					is.Equal(tt.wanted, err)
+				} else {
+					t.Errorf("SnapshotIterator.Teardown() error = %v, wantErr %v", err, tt.wantErr)
+				}
+			}
+		})
+	}
+}
 
+func TestSnapshotAtomicity(t *testing.T) {
+	is := is.New(t)
+	ctx := context.Background()
 	pool := test.ConnectPool(ctx, t, test.RegularConnString)
-	table := test.SetupTestTable(ctx, t, pool)
-	name := createTestSnapshot(t, pool)
 
-	conn, err := pool.Acquire(ctx)
-	is.NoErr(err)
-	s, err := NewSnapshotIterator(context.Background(), conn.Conn(), SnapshotConfig{
-		SnapshotName: name,
-		Table:        table,
-		Columns:      []string{"id", "key", "column1", "column2", "column3"},
-		KeyColumn:    "key",
-	})
+	// start our snapshot iterator
+	s, table := createTestSnapshotIterator(t, columns, key)
+	t.Cleanup(func() { is.NoErr(s.Teardown(ctx)) })
+	is.Equal(s.complete, false)
+
+	// add a record to our table after snapshot started
+	insertQuery := fmt.Sprintf(`INSERT INTO %s (id, column1, column2, column3)
+				VALUES (5, 'bizz', 456, false)`, table)
+	_, err := pool.Exec(ctx, insertQuery)
 	is.NoErr(err)
-	t.Cleanup(conn.Release)
+
+	// assert record does not appear in snapshot
+	for i := 0; i < 5; i++ {
+		r, err := s.Next(ctx)
+		if err != nil {
+			is.Equal(err, ErrSnapshotComplete)
+			is.Equal(r, sdk.Record{})
+			is.Equal(s.complete, true)
```

Comment:
Same comment about asserting private fields, all we care about is getting `ErrSnapshotComplete` from `Next`.